src/Simple.OData.Client.V4.Adapter/Extensions/ExtendedODataClient.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System.Collections.Generic;
+
+namespace Simple.OData.Client.V4.Adapter.Extensions
+{
+    /// <summary>
+    /// <inheritdoc cref="IExtendedODataClient"/>
+    /// </summary>
+    public class ExtendedODataClient : IExtendedODataClient
+    {
+        private readonly ODataClient _baseClient;
+
+        public ExtendedODataClient(ODataClient baseClient)
+        {
+            _baseClient = baseClient;
+        }
+
+        public IExtendedBoundClient<IDictionary<string, object>> For(string collectionName)
+        {
+            var client = new ExtendedBoundClient<IDictionary<string, object>>(_baseClient, _baseClient.Session);
+            client.For(collectionName);
+            _baseClient.Session.Settings.IgnoreUnmappedProperties = true;
+            _baseClient.Session.Settings.ReadUntypedAsString = false;
```

Comment:
I presume this works around some problem, but I'm not quite sure what, and it seems out of place.

Reply:
I have compeletely removed changing of session settings and added a remark to the xml docs to instruct users to set this setting if necessary